src/common/function/src/table/flush_compact_table.rs

```diff
@@ -34,78 +36,147 @@ use crate::ensure_greptime;
 use crate::function::{Function, FunctionContext};
 use crate::handlers::TableMutationHandlerRef;
 
-macro_rules! define_table_function {
-    ($name: expr, $display_name_str: expr, $display_name: ident, $func: ident, $request: ident) => {
-        /// A function to $func table, such as `$display_name(table_name)`.
-        #[admin_fn(name = $name, display_name = $display_name_str, sig_fn = "signature", ret = "uint64")]
-        pub(crate) async fn $display_name(
-            table_mutation_handler: &TableMutationHandlerRef,
-            query_ctx: &QueryContextRef,
-            params: &[ValueRef<'_>],
-        ) -> Result<Value> {
-            ensure!(
-                params.len() == 1,
-                InvalidFuncArgsSnafu {
-                    err_msg: format!(
-                        "The length of the args is not correct, expect 1, have: {}",
-                        params.len()
-                    ),
-                }
-            );
+#[admin_fn(
+    name = "FlushTableFunction",
+    display_name = "flush_table",
+    sig_fn = "flush_signature",
+    ret = "uint64"
+)]
+pub(crate) async fn flush_table(
+    table_mutation_handler: &TableMutationHandlerRef,
+    query_ctx: &QueryContextRef,
+    params: &[ValueRef<'_>],
+) -> Result<Value> {
+    ensure!(
+        params.len() == 1,
+        InvalidFuncArgsSnafu {
+            err_msg: format!(
+                "The length of the args is not correct, expect 1, have: {}",
+                params.len()
+            ),
+        }
+    );
 
-            let ValueRef::String(table_name) = params[0] else {
-                return UnsupportedInputDataTypeSnafu {
-                    function: $display_name_str,
-                    datatypes: params.iter().map(|v| v.data_type()).collect::<Vec<_>>(),
-                }
-                .fail();
-            };
-
-            let (catalog_name, schema_name, table_name) =
-                table_name_to_full_name(table_name, &query_ctx)
-                    .map_err(BoxedError::new)
-                    .context(TableMutationSnafu)?;
-
-            let affected_rows = table_mutation_handler
-                .$func(
-                    $request {
-                        catalog_name,
-                        schema_name,
-                        table_name,
-                    },
-                    query_ctx.clone(),
-                )
-                .await?;
-
-            Ok(Value::from(affected_rows as u64))
+    let ValueRef::String(table_name) = params[0] else {
+        return UnsupportedInputDataTypeSnafu {
+            function: "flush_table",
+            datatypes: params.iter().map(|v| v.data_type()).collect::<Vec<_>>(),
         }
+        .fail();
     };
+
+    let (catalog_name, schema_name, table_name) = table_name_to_full_name(table_name, query_ctx)
```

Comment:
make sure if the table_name is a full-qualified one, it shares same catalog name with query_ctx

Reply:
> make sure if the table_name is a full-qualified one, it shares same catalog name with query_ctx

Added catalog validation in this commit e48d09780.